test/e2e/util.go

```diff
@@ -3267,9 +3267,16 @@ func getNodePortURL(client *client.Client, ns, name string, svcPort int) (string
 	if err != nil {
 		return "", err
 	}
-	// It should be OK to list unschedulable Node here.
-	nodes, err := client.Nodes().List(api.ListOptions{})
-	if err != nil {
+	// This list of nodes must not include the master, which is marked
+	// unschedulable, since the master doesn't run kube-proxy. Without
+	// kube-proxy NodePorts won't work.
+	var nodes *api.NodeList
+	if wait.PollImmediate(poll, singleCallTimeout, func() (bool, error) {
+		nodes, err = client.Nodes().List(api.ListOptions{FieldSelector: fields.Set{
+			"spec.unschedulable": "false",
+		}.AsSelector()})
+		return err == nil, nil
+	}) != nil {
```

Comment:
You can also use https://github.com/kubernetes/kubernetes/blob/master/test/e2e/util.go#L2176